functions/assert_metadata_os.pp

```diff
@@ -0,0 +1,57 @@
+# Fails a compile if the client system is not compatible with the module's
+# metadata.json
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param match
+#   The level at which to match the release number
+#
+#   * none  -> No match on release (default)
+#   * full  -> Full release must match
+#   * major -> Only the major release must match
+#
+function simplib::assert_metadata_os (
```

Comment:
Can the function be called something a little less ambiguous? Maybe:
* `validate_supported_os`
* `on_supported_os`
* `check_metadata_for_os`